synapse/handlers/oidc_handler.py

```diff
@@ -674,6 +674,21 @@ async def handle_oidc_callback(self, request: SynapseRequest) -> None:
                 self._sso_handler.render_error(request, "invalid_token", str(e))
                 return
 
+        # first check if we're doing a UIA
+        if ui_auth_session_id:
+            try:
+                remote_user_id = self._remote_id_from_userinfo(userinfo)
+            except Exception as e:
```

Comment:
Should this only be `MappingException`?

Reply:
well, I don't think so.

We used to catch *all* exceptions (and turn them into  `MappingExceptions`) at https://github.com/matrix-org/synapse/pull/8881/files#diff-b70d930a35a860a3c89284899c7c71ead788407bb579dca9d2d2d82dd4cf321dR870-R873, so I'm just copying that. (In this case we can log the exception immediately, rather than turning it into a `MappingException` for a higher-level method to catch).

Reply:
Ah, makes sense. Just differs a bit from the SAML code is why I was asking. It seems fine. 👍